File: solutions/julia/bird-watcher/1/bird-watcher.jl
function today(birds_per_day)
    birds_per_day[end]
end

function increment_todays_count(birds_per_day)
    ans = [birds_per_day[i] for i in 1:(length(birds_per_day)-1)]
    push!(ans, birds_per_day[end] + 1)
    ans
end

function has_day_without_birds(birds_per_day)
    length([1 for b in birds_per_day if b == 0]) > 0
end

function count_for_first_days(birds_per_day, num_days)
    sum([birds_per_day[i] for i in 1:num_days])
end

function busy_days(birds_per_day)
    length([1 for b in birds_per_day if b >= 5])
end

function average_per_day(week1, week2)
    (week1 .+ week2) ./ 2.0
end

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        return new[] { 0, 2, 5, 3, 7, 8, 4};
    }

    public int Today()
    {
        return this.birdsPerDay.Last();
    }

    public void IncrementTodaysCount()
    {
        birdsPerDay[birdsPerDay.Length - 1] += 1;
    }

    public bool HasDayWithoutBirds()
    {
        return birdsPerDay.Any(x=>x==0);
    }

    public int CountForFirstDays(int numberOfDays)
    {
        var ans = 0;
        for (var i = 0; i < numberOfDays; i++)
        {
            ans += birdsPerDay[i];
        }
        return ans;
    }

    public int BusyDays()
    {
        return birdsPerDay.Count(x => x >= 5);
    }
}
